adapt_med_seg/data/dataset.py

```diff
@@ -49,54 +50,37 @@ def __init__(
     def name(self) -> str:
         return self._name
 
-    def __getitem__(self, idx) -> tuple[DataItem, torch.Tensor, str]:
+    def __getitem__(self, idx) -> tuple[DataItem, torch.Tensor, str, str]:
         ct_path = self._ct_paths[idx]
         gt_path = self._gt_paths[idx]
         modality = str(self._case_modality[idx])
+        label_map = self._case_label[idx]
+        # global is what we return as the label (e.g. pancreas)
+        global_label = self._labels[label_map["global_idx"]]
+        # local label is what we use to index the ground truth
+        local_label = int(label_map["local_idx"])
 
         ct_npy, gt_npy = self._processor.load_uniseg_case(ct_path, gt_path)
-        
+
+        # retrieve the mask, but keep the channel dimension
+        gt_npy = np.expand_dims(gt_npy[local_label-1,...], axis=0)
+
         ct_npy = ct_npy.astype("float32")
         gt_npy = gt_npy.astype("int64")
-        
+
         if self.train and idx not in getattr(self, "_test_indices", []):
             data_item = self._processor.train_transform(ct_npy, gt_npy)
         else:
             data_item = self._processor.zoom_transform(ct_npy, gt_npy)
 
-        return data_item, gt_npy, modality
+        # squeeze the channel dimension. it's always 1
+        data_item["label"] = data_item["label"].squeeze(0)
```

Comment:
This is a little concerning but if you say there are no issues with loading ok.

Reply:
The reason there can only be one is because on line 66, we select exactly one mask. The reason why we need to squeeze here is because SegVol's original logic was to select the class after the data loading happened (something like how we did it), although I have no idea how they trained that

Reply:
This was the easiest and most bugfree way to do this